src/server/coprocessor/executor/selection.rs

```diff
@@ -0,0 +1,284 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// FIXME(andelf): remove following later
+#![allow(dead_code)]
+
+use std::rc::Rc;
+use std::collections::HashSet;
+
+use tipb::executor::Selection;
+use tipb::schema::ColumnInfo;
+use tipb::expression::{Expr, ExprType};
+use util::xeval::{Evaluator, EvalContext};
+use util::codec::number::NumberDecoder;
+
+use super::{Row, Executor};
+use super::super::Result;
+use super::super::endpoint::inflate_with_col;
+
+/// Collect all mentioned column ids in expr tree
+pub struct ExprColumnRefVisitor {
```

Comment:
Is it share the same implementation with `use super::topn::ExprColumnRefVisitor;`?